sdks/go/pkg/beam/io/rtrackers/offsetrange/offsetrange_test.go

```diff
@@ -299,3 +301,299 @@ func TestTracker_TrySplit(t *testing.T) {
 		})
 	}
 }
+
+type offsetRangeEndEstimator struct {
+	EstimateRangeEnd int64
+}
+
+// Estimate provides the estimated end for unbounded offset range.
+func (o *offsetRangeEndEstimator) Estimate() int64 {
+	return o.EstimateRangeEnd
+}
+
+// SetEstimateRangeEnd sets the estimated end for unbounded offset range.
+func (o *offsetRangeEndEstimator) SetEstimateRangeEnd(rangeEnd int64) {
+	o.EstimateRangeEnd = rangeEnd
+}
+
+// TestNewGrowableTracker_Bad tests the behavior of NewGrowableTracker when wrong arguments are passed.
+func TestNewGrowableTracker_Bad(t *testing.T) {
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	_, err := NewGrowableTracker(rest, nil)
+	if err == nil {
+		t.Errorf("NewGrowableTracker() expected to throw error.")
+	}
+}
+
+// TestGrowableTracker_TryClaim tests the TryClaim method for GrowableTracker.
+func TestGrowableTracker_TryClaim(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+
+	if !tracker.TryClaim(int64(10)) {
+		t.Errorf("tracker.TryClaim(10) = %v, want: %v", false, true)
+	}
+	if !tracker.TryClaim(int64(100)) {
+		t.Errorf("tracker.TryClaim(10) = %v, want: %v", false, true)
+	}
+	if tracker.TryClaim(int64(math.MaxInt64)) {
+		t.Errorf("tracker.TryClaim(math.MaxInt64) = %v, want: %v, %v", true, false, tracker.err)
+	}
+	if tracker.IsDone() {
+		t.Errorf("tracker has done all work, but IsDone() returns false")
+	}
+}
+
+// TestGrowableTracker_SplitBeforeStart tests TrySplit() method for GrowableTracker
+// before claiming anything.
+func TestGrowableTracker_SplitBeforeStart(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	estimator.SetEstimateRangeEnd(10)
+	p, r, _ := tracker.TrySplit(0)
+
+	expected := Restriction{0, 0}
```

Comment:
nit: prefer `want` instead of expected.

Similarly, you could name the returned restrictions as gotP, and gotR, which signals to readers that these are the variables that we'll be comparing to soon.